src/image_processing.rs

```diff
@@ -128,11 +172,75 @@ impl ARAwareTransform {
         };
 
         if let Some(target_size) = self.aspect_ratio_to_size.get(&aspect_ratio) {
-            image.resize_to_fill(
-                target_size.0 as u32,
-                target_size.1 as u32,
-                image::imageops::FilterType::Lanczos3,
-            ) // returns the resized image
+            // Check if resize is actually needed
+            if image.width() == target_size.0 && image.height() == target_size.1 {
+                image.clone()
+            } else {
+                let image_pixel_type = image.pixel_type().unwrap();
+                if image_pixel_type == fr::PixelType::U8
+                    || image_pixel_type == fr::PixelType::U8x2
+                    || image_pixel_type == fr::PixelType::U8x3
+                    || image_pixel_type == fr::PixelType::U8x4
+                {
```

Comment:
this is the happy path: pulling in fast_image_resize, which is much faster than the image crate at resizing